apps/meteor/ee/client/views/admin/deviceManagement/DeviceManagementAdminRoute.tsx

```diff
@@ -1,14 +1,61 @@
-import { usePermission } from '@rocket.chat/ui-contexts';
+import { usePermission, useRouter, useSetModal, useTranslation } from '@rocket.chat/ui-contexts';
 import type { ReactElement } from 'react';
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 
+import GenericUpsellModal from '../../../../../client/components/GenericUpsellModal';
+import PageSkeleton from '../../../../../client/components/PageSkeleton';
+import { useIsEnterprise } from '../../../../../client/hooks/useIsEnterprise';
 import NotAuthorizedPage from '../../../../../client/views/notAuthorized/NotAuthorizedPage';
+import { useHasLicenseModule } from '../../../hooks/useHasLicenseModule';
 import DeviceManagementAdminPage from './DeviceManagementAdminPage';
 
 const DeviceManagementAdminRoute = (): ReactElement => {
+	const t = useTranslation();
 	const canViewDeviceManagement = usePermission('view-device-management');
 
-	if (!canViewDeviceManagement) {
+	const { data } = useIsEnterprise();
+	const hasDeviceManagement = useHasLicenseModule('engagement-dashboard');
+	const isUpsell = !data?.isEnterprise || !hasDeviceManagement;
+
+	const router = useRouter();
+
+	const setModal = useSetModal();
+	const [isModalOpen, setIsModalOpen] = useState(false);
+
+	const handleOpenModal = useCallback(() => {
+		setModal(
+			<GenericUpsellModal
+				title={t('Device_Management')}
+				img='images/device-management.png'
+				subtitle={t('Ensure_secure_workspace_access')}
+				description={t('Manage_which_devices')}
+				onCloseEffect={() => setIsModalOpen(false)}
+			/>,
+		);
+		setIsModalOpen(true);
+	}, [setModal, t]);
+
+	useEffect(() => {
+		router.subscribeToRouteChange(() => {
+			if (router.getRouteName() !== 'device-management') {
+				return;
+			}
+
+			if (isUpsell) {
+				handleOpenModal();
+			}
+		});
```

Comment:
All "subrouting" (i.e. handling of paths like `/admin/device-management/*`) is made by **one** route, named `device-management`. Therefore we don't need to watch for route changes.

```suggestion
		if (isUpsell) {
			handleOpenModal();
		}
```